spec/models/claim/cleaner_spec.rb

```diff
@@ -0,0 +1,75 @@
+require "rails_helper"
+
+RSpec.describe Claim::Cleaner, type: :model do
+
+  let(:over_two_months_ago) { 2.months.ago - 1.day }
+
+  it "does not delete details from a newly submitted claim" do
+    claim = create(:claim, :submitted)
+
+    Claim::Cleaner.new.delete_pii
+    cleaned_claim = Claim.find(claim.id)
+
+    expect(cleaned_claim.attributes).to eq(claim.attributes)
+  end
+
+  it "does not delete details from an approved but unpaid claim" do
+    claim = create(:claim, :approved)
+
+    Claim::Cleaner.new.delete_pii
+    cleaned_claim = Claim.find(claim.id)
+
+    expect(cleaned_claim.attributes).to eq(claim.attributes)
+  end
+
+  it "deletes details from an old rejected claim" do
+    claim = create(:claim, :rejected, updated_at: over_two_months_ago)
+
+    Claim::Cleaner.new.delete_pii
+    cleaned_claim = Claim.find(claim.id)
+
+    expect(cleaned_claim.attributes).to_not eq(claim.attributes)
+  end
+
+  it "deletes details from an old paid claim" do
+    claim = create(:claim, :approved)
+    create(:payment, :with_figures, claims: [claim], scheduled_payment_date: over_two_months_ago)
+
+    Claim::Cleaner.new.delete_pii
+    cleaned_claim = Claim.find(claim.id)
+
+    expect(cleaned_claim.attributes).to_not eq(claim.attributes)
+  end
+
+  it "deletes the fields we expect" do
+    claim = create(:claim, :rejected, updated_at: over_two_months_ago)
+
+    Claim::Cleaner.new.delete_pii
+    cleaned_claim = Claim.find(claim.id)
+
+    expect(cleaned_claim.first_name).to be_nil
+    expect(cleaned_claim.middle_name).to be_nil
+    expect(cleaned_claim.surname).to be_nil
+    expect(cleaned_claim.date_of_birth).to be_nil
+    expect(cleaned_claim.address_line_1).to be_nil
+    expect(cleaned_claim.address_line_2).to be_nil
+    expect(cleaned_claim.address_line_3).to be_nil
+    expect(cleaned_claim.address_line_4).to be_nil
+    expect(cleaned_claim.postcode).to be_nil
+    expect(cleaned_claim.payroll_gender).to be_nil
+    expect(cleaned_claim.national_insurance_number).to be_nil
+    expect(cleaned_claim.bank_sort_code).to be_nil
+    expect(cleaned_claim.bank_account_number).to be_nil
+    expect(cleaned_claim.building_society_roll_number).to be_nil
+  end
+
+  it "records when the claim had PII removed" do
```

Comment:
I'd probably merge this test with the previous one as we're testing the same part of the code

Reply:
I think it's describing two separate behaviours (one that the fields are actually nilled, the second that it records when that happened). The fact the class does that in one update is an implementation detail.